Radio Page

drigato opened the pull request.

drigato committed:
Stopped redirecting to /login if no valid session

drigato committed:
SAAS-728: Make schedule widgets embeddable

drigato committed:
Merge branch 'saas' into saas-landing-page

drigato committed:
SAAS-732: Clean up weekly program widget

drigato committed:
Separated weekly schedule widget css into separate file

drigato committed:
Merge branch 'saas' into saas-landing-page

drigato committed:
SAAS-762: Implement schedule widget design

drigato committed:
SAAS-786: Implement Landing Page Design

Implemented logo, station name and description

drigato committed:
SAAS-786: Implement Landing Page Design

- Separated embed player css from radio page css because I had to change
the responsive design a bit.
- Trying to embed the iframe on the radio page but there are some
conflicting styles

drigato committed:
SAAS-786: Implement Landing Page Design

Fixed player iframe to bottom of screen

drigato committed:
SAAS-811: Insert extra radio page buttons onto player bar

drigato committed:
Small fix to embed player metadata

drigato committed:
SAAS-812: Remove live chat from radio page

drigato committed:
Removed extra white space

drigato committed:
SAAS-813: Wire up schedule widget on Radio Page

drigato committed:
Merge branch 'saas-dev' into saas-landing-page

Conflicts:
	airtime_mvc/application/controllers/ApiController.php
	airtime_mvc/application/views/scripts/embed/player.phtml

drigato committed:
SAAS-821: Wire up new player bar with muses

Play and stop is working

drigato committed:
SAAS-821: Wire up new player bar with muses

Got metadata displaying

drigato committed:
SAAS-811: Insert extra radio page buttons onto player bar

Insert extra buttons onto player bar once the iframe has loaded

drigato committed:
SAAS-814: Decide what to do with new Login Page

We decided to link to the existing login page for now

drigato committed:
SAAS-833: Radio page UI improvements/fixes

Removed unnecessary css classes from premium player css file

drigato committed:
SAAS-833: Radio page UI improvements/fixes

Removed unnecessary/duplicate css classes from radio page css file

drigato committed:
SAAS-833: Radio page UI improvements/fixes

Fix schedule widget responsiveness

drigato committed:
SAAS-833: Radio page UI improvements/fixes

Removed scrolling on entire page and added it only to schedule widget

drigato committed:
SAAS-833: Radio page UI improvements/fixes

Fix spacing between schedule widget and logo

drigato committed:
SAAS-833: Radio page UI improvements/fixes

Moved player bar buttons to float right

drigato committed:
SAAS-833: Radio page UI improvements/fixes

Fixed schedule widget scrolling
Inserted default station description

drigato committed:
SAAS-833: Radio page UI improvements/fixes

Added width and height to schedule widget iframe tag

drigato committed:
SAAS-833: Radio page UI improvements/fixes

Undo some previous css changes

drigato committed:
Unhardcode premium style player

Had this is for testing purposes

drigato committed:
Remove unused embed controller action

drigato committed:
SAAS-841: Default station logo to the Airtime logo

drigato committed:
SAAS-841: Default station logo to the Airtime logo

logo image file

drigato committed:
SAAS-842: Schedule widget show time formatting problem

drigato committed:
SAAS-843: Create Embeddable Schedule Widget View

Replaced player controller with embeddablewidgets controller.
Replaced 'Player' page with 'Widgets' page.
Added schedule widget to the widgets page.

drigato committed:
Fix typo

drigato committed:
SAAS-836: Create a new schedule widget skin

drigato committed:
SAAS-851: Embed schedule widget loses format with long show names

drigato committed:
SAAS-851: Embed schedule widget loses format with long show names

Applied fix for premium style sheet

drigato committed:
Merge branch 'saas-dev' into saas-landing-page

Conflicts:
	airtime_mvc/application/controllers/ApiController.php
	airtime_mvc/application/controllers/IndexController.php

drigato committed:
SAAS-855: Default station description in Preferences

drigato committed:
Fix schedule widget layout by removing table structure

asantoni commented on airtime_mvc/application/Bootstrap.php:
```diff
@@ -133,7 +133,11 @@ protected function _initHeadLink()
+        // Don't include style.css on the Radio Page because there are conflicting styles.
+        if ($_SERVER['REQUEST_URI'] != "/") {
+            $view->headLink()->appendStylesheet($baseUrl . 'css/styles.css?' . $CC_CONFIG['airtime_version']);
+        }
```
This is a nasty hack. Is there not a better way to remove this explicitly from one page? 

In the long-term, these stylesheets should not even be included in the Bootstrap file. We should have some sort appendDefaultStylesheets() function that's injected in our controllers.

asantoni commented on airtime_mvc/application/common/WidgetHelper.php:
A) These hardcoded day of week constants could be generated with date().
B) The API is not for human consumption, so outputting an array of labelled days seems weird. They don't need explicit labels if we're always returning Monday to Sunday. It could just be an array of 7 dictionaries.
C) I think we should consider always returning 7 days from the current day. If it's a Sunday, what good does showing me the past 6 days do?

asantoni commented on airtime_mvc/application/common/WidgetHelper.php:
Make the 7 a constant, like DAYS_PER_WEEK. Also try to avoid the use of $i and make your iterables more readable. If $i is the day of the week, just call it $dayOfWeek. It helps with readability in longer for-loops like this.

asantoni commented on airtime_mvc/application/controllers/EmbedController.php:
```diff
@@ -54,4 +63,43 @@ public function playerAction()
+            $this->view->widgetStyle = "basic";
+            $this->view->css = Application_Common_HTTPHelper::getStationUrl() . "/css/embed/weekly-schedule-widget-basic.css?" . $CC_CONFIG['airtime_version'];
+        }
+        $this->view->jquery = Application_Common_HTTPHelper::getStationUrl() . "widgets/js/jquery-1.6.1.min.js?".$CC_CONFIG['airtime_version'];
```
This is another thing we should fix at some point. Hardcoding jquery seems bad. Would be good if we had a factory for all these includes...

asantoni commented on airtime_mvc/application/controllers/EmbedController.php:
```diff
@@ -54,4 +63,43 @@ public function playerAction()
+
+        $result = WidgetHelper::getWeekInfoV2($this->getRequest()->getParam("timezone"));
+        $this->view->scheduleDataWeek1 = $result[0];
+        $this->view->scheduleDataWeek2 = $result[1];
```
Why use two separate variables instead of an array?

danielhjames commented on airtime_mvc/application/common/WidgetHelper.php:
Day of the week labels might be useful for debugging API problems, since Airtime allows the user to set 'Week Starts On:' in the Preferences page.
Displaying previous days is only useful in combination with the API exporting a show URL pointing to a page where the show can be downloaded as a file, i.e. as a catch-up service. In that circumstance, users probably want the seven days ahead plus the seven days behind. BBC Radio is now offering up to one month behind for catch-up.

asantoni commented on airtime_mvc/application/controllers/EmbeddablewidgetsController.php:
```diff
@@ -20,12 +21,17 @@ public function customizeAction()
+            $this->view->player_form = $form;
         } else {
-            $this->view->errorMsg = "To configure and use the embeddable player you must:<br><br>
+            $this->view->player_error_msg = "To configure and use the embeddable player you must:<br><br>
```
This string should be wrapped in _pro()

asantoni commented on airtime_mvc/application/controllers/EmbeddablewidgetsController.php:
```diff
@@ -20,12 +21,17 @@ public function customizeAction()
         }
 
+        if (!$apiEnabled) {
+            $this->view->weekly_schedule_error_msg = "To use the embeddable weekly schedule you must:<br><br>
```
Same as above, wrap in _pro()

asantoni commented on airtime_mvc/application/layouts/scripts/radio-page.phtml:
```diff
@@ -0,0 +1,36 @@
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link href='http://fonts.googleapis.com/css?family=Roboto:400,100,300,700' rel='stylesheet' type='text/css'>
+
+    <script src="http://ajax.googleapis.com/ajax/libs/jquery/1/jquery.min.js"></script>
```
Is this not a second copy of jquery??

asantoni commented on airtime_mvc/application/layouts/scripts/radio-page.phtml:
```diff
@@ -0,0 +1,36 @@
+        $companySiteAnchor);
+    ?>
+</div>
+-->
```
Remove the commented code if you're done

asantoni commented on airtime_mvc/application/models/Preference.php:
```diff
@@ -498,7 +498,12 @@ public static function SetStationDescription($description)
+        if (!empty($description)) {
+            return $description;
+        } else {
+            return "Powered by Airtime Pro";
```
_pro() translation wrapper? (Maybe? This is a weird case...)

drigato replied:
Maybe we should only translate the "Powered by" part?

asantoni replied:
Yeah. "Airtime Pro" should be one of the branding constants too, with that interpolated into it with sprintf.

danielhjames replied:
From a localization point of view I would suggest translating the whole string, because in some languages the word order might be: 'Airtime Pro powered by' even if you do not translate 'Airtime Pro' itself. You might also want the flexibility to insert another brand name here for a white-label client, by changing only the .po file.

asantoni replied:
This is what sprintf is for. You put the % placeholder in the translatable string and that allows the translator to place the "Airtime Pro" string correctly according to their grammar.

danielhjames replied:
Using a placeholder is better than nothing, although then the translator might need to be provided with some context. For example 'Powered by %' in the case of Airtime Pro is an idiom and might be difficult to translate, compared to say 'Powered by % lithium-ion batteries' where % is the brand of batteries.

asantoni replied:
Good translation frameworks give you the option to provide context or
disambiguation in your translation macros. I believe gettext does this if
you add the pipe symbol  | to your strings. We may need to be running
gettext with some extra argument as well.

On Wed, Jun 10, 2015 at 12:39 PM, Daniel James notifications@github.com
wrote:

> In airtime_mvc/application/models/Preference.php
> https://github.com/sourcefabric/airtime/pull/116#discussion_r32137799:
> 
> > @@ -498,7 +498,12 @@ public static function SetStationDescription($description)
> > 
> > ```
> >  public static function GetStationDescription()
> >  {
> > ```
> > -        return self::getValue("description");
> > -        $description = self::getValue("description");
> > -        if (!empty($description)) {
> > -            return $description;
> > -        } else {
> > -            return "Powered by Airtime Pro";
> 
> Using a placeholder is better than nothing, although then the translator
> might need to be provided with some context. For example 'Powered by %' in
> the case of Airtime Pro is an idiom and might be difficult to translate,
> compared to say 'Powered by % lithium-ion batteries' where % is the brand
> of batteries.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/sourcefabric/airtime/pull/116/files#r32137799.

## 

Albert Santoni
Airtime Lead Developer, Sourcefabric
albert.santoni@sourcefabric.org

www.sourcefabric.org

720 Bathurst St.
Suite 203
Toronto, ON / M5S 2R4 / Canada

danielhjames replied:
https://core.trac.wordpress.org/ticket/9112 describes both context and comment lines in .po files. We would need to test if these show up in the Transifex interface.

asantoni commented on airtime_mvc/application/models/Preference.php:
```diff
@@ -602,7 +607,14 @@ public static function SetStationLogo($imagePath)
+        } else {
+            // We return the Airtime logo if no logo is set in the database.
+            // airtime_logo.png is stored under the public directory
+            return "airtime_logo.png";
```
This is misleading because in airtime_mvc/application/views/scripts/form/preferences_general.phtml you're actually setting the image source to "airtime_logo.png" conditionally. (You've repeated yourself here...)

By reading the code above, I would expect that changing the "airtime_logo.png" string to "foobar.png" would make the default logo point to "foobar.png", but the code in preferences_general.phtml doesn't work like that.

I didn't realize that this function had to return a base64-encoded image. In this case, I think it'd be better to return a magic constant indicating that the default station logo should be used, and then also set a constant for that. (See my comments below in preferences_general.phtml for clarification).

Specifically, on line 616 above I'd return DEFAULT_LOGO_PLACEHOLDER, and define that in constants.php.

asantoni commented on airtime_mvc/application/views/scripts/form/preferences_general.phtml:
```diff
@@ -12,7 +12,15 @@
+            <?php
+            $logoImg = $this->element->getView()->logoImg;
+            if ($logoImg === "airtime_logo.png") {
+                $src = "airtime_logo.png";
```
Continuing my comment from above... I'd change this logic to be like, if ($logoImg === DEFAULT_LOGO_PLACEHOLDER) then $src = DEFAULT_LOGO_FILE.

where DEFAULT_LOGO_FILE would be a constant defined to "airtime_logo.png" in constants.php.

drigato committed:
Merge branch 'saas-dev' into saas-landing-page

drigato committed:
Radio Page Code Review Changes

Reset stylesheets in index controller.
Removed hardcoded days of week in WidgetHelper.
Display schedule widget with 6 days after the current day.

drigato committed:
Merge branch 'saas-dev' into saas-landing-page
